galasa-ui/src/app/layout.tsx

```diff
@@ -13,32 +13,36 @@ import { FeatureFlagProvider } from '@/contexts/FeatureFlagContext';
 import { cookies } from 'next/headers';
 import FeatureFlagCookies from '@/utils/featureFlagCookies';
 import { getLocale } from 'next-intl/server';
+import { ThemeProvider } from '@/contexts/ThemeContext';
 
 export const dynamic = "force-dynamic";
+type ThemeType = "white" | "g100"; 
 
-export default async function RootLayout({children,}: {children: React.ReactNode}) {
-  // Ensure that the incoming `locale` is valid
+export default async function RootLayout({ children }: { children: React.ReactNode }) {
   const locale = await getLocale();
- 
+
   const galasaServiceName = process.env.GALASA_SERVICE_NAME?.trim() || "Galasa Service";
   const featureFlagsCookie = cookies().get(FeatureFlagCookies.FEATURE_FLAGS)?.value;
+  const themeCookie = cookies().get('preferred-theme')?.value;
```

Comment:
Can we avoid using a cookie for the preferred theme?

Cookies are sent to the server on every request, which isn't ideal. I think theme-related config should be stored client-side in localStorage.